customcommands/web.go

```diff
@@ -539,6 +588,117 @@ func handleUpdateAndRunNow(w http.ResponseWriter, r *http.Request) (web.Template
+	}
+
+	if !premium.ContextPremium(ctx) && cmdEdit.TriggerOnEdit {
+		return templateData.AddAlerts(web.ErrorAlert("`Trigger on edits` is a premium feature, your command wasn't saved, please save again after disabling `Trigger on edits`")), nil
+	}
+
+	dbModel := cmdEdit.ToDBModel()
+
+	templateData["CurrentGroupID"] = dbModel.GroupID.Int64
+
+	dbModel.GuildID = activeGuild.ID
+	dbModel.LocalID = cmdEdit.ID
+	dbModel.TriggerType = int(triggerTypeFromForm(cmdEdit.TriggerTypeForm))
+	// check low interval limits
+	if dbModel.TriggerType == int(CommandTriggerInterval) && dbModel.TimeTriggerInterval <= 10 {
+		if dbModel.TimeTriggerInterval < 5 {
+			dbModel.TimeTriggerInterval = 5
+		}
+
+		ok, err := checkIntervalLimits(ctx, activeGuild.ID, dbModel.LocalID, templateData)
+		if err != nil || !ok {
+			return templateData, err
+		}
+	}
+
+	blacklistColumns := []string{"last_run", "next_run", "local_id", "guild_id", "last_error", "last_error_time", "run_count", "import_count"}
+	if cmdSaved.PublicID == "" {
+		hash := sha1.New()
+		io.WriteString(hash, strconv.FormatInt(activeGuild.ID, 10))
+		io.WriteString(hash, strconv.FormatInt(time.Now().Unix(), 10))
+		fullHash := base64.URLEncoding.EncodeToString(hash.Sum(nil))
+
+		// baseIDLenth is the minimum length for a public ID
+		const baseIDLength = 5
+		truncatedHash := addCharIfCollides(ctx, fullHash, baseIDLength)
+		dbModel.PublicID = truncatedHash
+		dbModel.Public = true
+		templateData["PublicLink"] = getPublicLink(dbModel)
+	} else {
+		blacklistColumns = append(blacklistColumns, "public_id")
+	}
+
+	_, err = dbModel.UpdateG(ctx, boil.Blacklist(blacklistColumns...))
+	if err != nil {
+		logger.Error(errors.WithStackIf(err))
+		return templateData, nil
+	}
+
+	// create, update or remove the next run time and scheduled event
+	if dbModel.TriggerType == int(CommandTriggerInterval) {
+		// need the last run time
+		fullModel, err := models.CustomCommands(qm.Where("guild_id = ? AND local_id = ?", activeGuild.ID, dbModel.LocalID)).OneG(ctx)
+		if err != nil {
+			web.CtxLogger(ctx).WithError(err).Error("failed retrieving full model")
+		} else {
+			err = UpdateCommandNextRunTime(fullModel, true, true)
+		}
+	} else {
+		err = DelNextRunEvent(activeGuild.ID, dbModel.LocalID)
+	}
+
+	if err != nil {
+		web.CtxLogger(ctx).WithError(err).WithField("guild", dbModel.GuildID).Error("failed updating next custom command run time")
+	}
+
+	go cplogs.RetryAddEntry(web.NewLogEntryFromContext(r.Context(), panelLogKeyUpdatedCommand, &cplogs.Param{Type: cplogs.ParamTypeInt, Value: dbModel.LocalID}))
+
+	pubsub.EvictCacheSet(cachedCommandsMessage, activeGuild.ID)
+
+	return templateData, err
+}
+
+// checks the ID truncated at length characters, and returns this truncated
+// string if it doesn't already exist. If it does exist, the function runs
+// recursively, adding 1 to length and returns the result
+func addCharIfCollides(ctx context.Context, full string, length int) string {
+	truncated := full[:length]
```

Comment:
Possible YAGNI, the chances of collision here are negligible, if you think this can happen maybe the code can be 10 chars long by default. git uses 7 char long truncated hashes, so 10 should be sufficient.